further_ml_examples/design_patterns/dp5_ch3_reframing.ipynb

```diff
@@ -0,0 +1,172 @@
+    "\n",
+    "<div>\n",
+    "<center><img src=\"Images/reframing/output_usage.png\" width=\"700\"/></center>\n",
+    "</div>\n",
+    "\n",
+    "A further compramise is the lack of  comparison models when a problem is reframed. Comparision between a regression and classification model is not straight forward.\n",
+    "\n",
+    "# Best of Both Worlds\n",
+    "\n",
+    "Maybe you can reframe your problem as 2 problems with a similar goal to get the best of both worlds in the end. The book mentions both hard weight sharing (sharing some layers) and soft weight sharing (including weight similarity in the loss, W <sub>penalty</sub>), allowing you to leverage both the probabalistic and  \n",
+    "\n",
+    "<div>\n",
+    "<center><img src=\"Images/reframing/weight_sharing.png\" width=\"700\"/></center>\n",
+    "</div>\n",
+    "\n",
+    "\n",
+    "## Summary\n",
+    "\n",
+    "Although this chapter introduces alot of seperate ideas, maybe it can be 'reframed' as being concious of the choices you are making throughout the design process. Particualry your 'input(s)', 'question(s)' and 'output(s)', these aspects of your problem can often be tweaked in order to benifit your overall system.\n",
+    "\n",
+    "<div>\n",
+    "<center><img src=\"Images/reframing/summary.png\" width=\"700\"/></center>\n",
+    "</div>\n",
+    "\n",
+    "- Inputs - What is  nature of the data you want to present to the mode?\n",
+    "    - Time of day - Is the time important or is it the point on the cycle?!\n",
+    "    - Textual - Some sort of embedding previosuly discussed?!\n",
+    "- Question - Can we make the problem any simpler?\n",
+    "    - Regression / Classification?!\n",
+    "- Outputs - What is useful to the end user of the system?\n",
+    "    - Is the output probabilistic?\n",
+    "    \n",
+    "## Extra Credit\n",
+    "\n",
+    "For something really at the extreme end of reframing, look at the area of self-supervised learning which ties together many of the ideas touched upon in the book so far. A nice starting point is the paper: [Unsupervised Visual Representation Learning by Context Prediction, Doresh et Al ICCV 2015](https://ieeexplore.ieee.org/document/7410524). It presents the idea of learning some task that is trivial to generate labels for, with the goal to learn some representation / embedding along the way that encodes something useful for a different task, such as object recogni\n",
+    "\n",
+    "\n",
+    "<div>\n",
+    "<center><img src=\"Images/reframing/pretext.png\" width=\"700\"/></center>\n",
+    "</div>\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5b8f07e1-7cb6-4958-bcd5-631d904e9620",
+   "metadata": {},
+   "source": [
+    "## Example python implementation"
```

Comment:
This all looks great @owena11 . You might want to remove the markdown titles in the notebook with the missing examples, but otherwise looks good to merge